### PR TITLE
Add basic integration and e2e tests, env setup

### LINKED ISSUE #674: Add operator E2E and integration tests
It would be nice to have operator end-2-end and integration tests to validate at least basic reconcile logic.
Also complex Tekton deployment logic can be implemented with real operator build and deploy.

### PR DESCRIPTION
**Pull request by dmvolod**
Fixes #674
Blocked by #673

### CODE REVIEW

File: karavan-operator/src/main/java/org/apache/camel/karavan/operator/resource/PipelineRoleDeployer.java [hunk 1]
```diff
@@ -42,7 +42,7 @@ public Role desired(Karavan karavan, Context<Karavan> context) {
                 .withRules(
                         new PolicyRuleBuilder().withApiGroups("").withResources("secrets", "configmaps", "services", "persistentvolumes", "persistentvolumeclaims").withVerbs("*").build(),
                         new PolicyRuleBuilder().withApiGroups("networking.k8s.io").withResources("ingresses").withVerbs("*").build(),
-                        new PolicyRuleBuilder().withApiGroups("route.openshift.io").withResources( "routes").withVerbs("*").build(),
+                        new PolicyRuleBuilder().withApiGroups("route.openshift.io").withResources( "routes").withResources( "routes/custom-host").withVerbs("*").build(),
```
Comment (mgubaidullin) [inline karavan-operator/src/main/java/org/apache/camel/karavan/operator/resource/PipelineRoleDeployer.java:45]: What "routes/custom-host" mean?

File: karavan-operator/src/main/java/org/apache/camel/karavan/operator/resource/KaravanRole.java [hunk 2]
```diff
@@ -43,7 +43,7 @@ public Role desired(Karavan karavan, Context<Karavan> context) {
                         new PolicyRuleBuilder().withApiGroups("").withResources("secrets", "configmaps").withVerbs("get", "list").build(),
                         new PolicyRuleBuilder().withApiGroups("").withResources("persistentvolumes", "persistentvolumeclaims").withVerbs("get", "list", "watch").build(),
                         new PolicyRuleBuilder().withApiGroups("tekton.dev").withResources("pipelineruns").withVerbs("*").build(),
-                        new PolicyRuleBuilder().withApiGroups("").withResources("pods", "services", "routes", "replicationcontrollers").withVerbs("*").build(),
```
[the remark this answers is not in the document]
Comment (dmvolod) [reply karavan-operator/src/main/java/org/apache/camel/karavan/operator/resource/KaravanRole.java:46]: Ok, will add the same permission as for `Constants.ROLE_PIPELINE_DEPLOYER`

File: karavan-operator/src/main/java/org/apache/camel/karavan/operator/resource/PipelineRoleDeployer.java (hunk 1 above)
Comment (dmvolod) [reply karavan-operator/src/main/java/org/apache/camel/karavan/operator/resource/PipelineRoleDeployer.java:45]: Probably it should be omitted, this is additional permission to set `host: def.abc.com` field.
However, to provide customization ability of the `service/route/ingress` objects, corresponding fields should be specified in `Karavan` CRD in hierarchical manner, not plain one.

File: karavan-operator/src/main/java/org/apache/camel/karavan/operator/resource/KaravanRole.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (dmvolod) [reply karavan-operator/src/main/java/org/apache/camel/karavan/operator/resource/KaravanRole.java:46]: Fixed

**Comment by mgubaidullin** [comment]
@dmvolod looks like we have operators tests in k8s! Thanks!

**Pull request merged**